src/main/kotlin/no/nav/syfo/client/graphapi/GraphApiClient.kt

```diff
@@ -0,0 +1,73 @@
+package no.nav.syfo.client.graphapi
+
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.httpClientProxy
+import no.nav.syfo.util.bearerHeader
+import org.slf4j.LoggerFactory
+
+class GraphApiClient(
+    private val azureAdClient: AzureAdClient,
+    private val baseUrl: String,
+    private val adRollerWeCareAbout: List<AdRolle>
+) {
+    private val httpClient = httpClientProxy()
+
+    suspend fun hasAccess(
+        // TODO: legg inn callId
+        adRolle: AdRolle,
+        token: String,
+    ): Boolean {
+        val groupList = getRoleList(token = token)
+
+        return isRoleInUserGroupList(
+            groupList = groupList.value,
+            adRolle = adRolle,
+        )
+    }
+
+    suspend fun getRoleList(token: String): GraphApiUserGroupsResponse {
+        // TODO: sjekk cachen først
+        // TODO: add callId for exception handling
+        // TODO: add metrics
+        val oboToken = azureAdClient.getOnBehalfOfToken(
+            scopeClientId = baseUrl,
+            token = token,
+        )?.accessToken ?: throw RuntimeException("Failed to request access to Enhet: Failed to get OBO token")
```

Comment:
Er det access to Enhet man ber om her? 💭

Reply:
Nope, fixer!